internal/charmstore/store.go

```diff
@@ -570,9 +570,30 @@ func (s *Store) findEntityInChannel(url *charm.URL, ch params.Channel, fields ma
 	}
 	var entityURL *charm.URL
 	if url.Series == "" {
-		for _, u := range baseEntity.ChannelEntities[ch] {
-			if entityURL == nil || seriesScore[u.Series] > seriesScore[entityURL.Series] {
+		var entitySeries string
+		for s, u := range baseEntity.ChannelEntities[ch] {
+			if entityURL == nil {
 				entityURL = u
+				entitySeries = s
+				continue
+			}
+			switch seriesCmp(u.Series, entityURL.Series) {
```

Comment:
I'm not really sure that seriesCmp is worth it, as seriesScore[s] is quite
understandable from my p.o.v.

Perhaps just:

    d := seriesScore[u.Series] - seriesScore[entityURL.Series]
    switch {
    case d > 0:
         // Different series in the URLs and u.Series is better.
    case d == 0:
         // Because each ... etc
         if seriesScore[s] > seriesScore[entitySeries] {

or...

tbh I didn't mind the original as it made it clear that it was a simple
maximum test (albeit with some somewhat initially-hard-to-understand
conditions).

My understanding difficulties weren't from issues in comparing series,
but from which series were being compared and when.

A comment would have made all the difference (the "Because each"
command below is nice). I'm not sure the longer
code is that much clearer.